src/Components/MissionPage/ActiveMission/ActiveMission.tsx

```diff
@@ -8,18 +8,42 @@ import axios from "axios";
 import MissionSummaryDialog from "./MissionSummaryDialog";
 import FinishMissionButton from "./FinishMissionButton";
 import RedeemRewardButton from "~/Components/MissionPage/ActiveMission/RedeemRewardBotton";
+import UserData from "~/types/Response/UserData";
+import { useNavigate } from "react-router";
+import MissionResponse from "~/types/Response/MissionResponse";
+import ChoreResponse from "~/types/Response/ChoreResponse";
+import RewardResponse from "~/types/Response/RewardResponse";
 
 interface ActiveMissionProps {
   missionChores: MissionChoreResponse[];
+  userData: UserData;
+  setUserData: (
+    userData: (prevData: UserData) => {
+      userId: number;
+      firstName: string;
+      lastName: string;
+      email: string;
+      username: string;
+      chores: ChoreResponse[];
+      missions: MissionResponse[];
+      rewards: RewardResponse[];
+      message: string | null;
+    },
```

Comment:
You don't have to change this up, FYI, you can rewrite this as 
```suggestion
    userData: Dispatch<SetStateAction<UserData>>;
```